app/controllers/stories_controller.rb

```diff
@@ -329,6 +329,22 @@ def unsave
     render :plain => "ok"
   end
 
+  def check_url_dupe
+    @story = Story.new(story_params)
+
+    if @story.url.present?
+      @story.check_url
+
+      # ignore other types of errors (e.g., invalid URL format)
+      if @story.already_posted_story.blank?
```

Comment:
Instead of extracting all of `check_url` from `validate`, extract just the part that sets `already_posted_story` and maybe adds the error about it being submitted recently. Leave the regexp that checks format and adds that error.

I don't want this piece of code to have to know about all the other possible errors on URL and be maintained in sync with the model.